tests/server_spec.rs

```diff
@@ -0,0 +1,36 @@
+use tailcall::config::Config;
+use tailcall::http::start_server;
+
+#[tokio::test]
+async fn server_start() {
+  use reqwest::Client;
+  use serde_json::json;
+
+  tokio::spawn(async {
+    let file_paths = vec!["tests/server/config/server-start.graphql".to_string()];
+    let config = Config::from_file_or_url(file_paths.iter()).await.unwrap();
+    start_server(config).await.unwrap();
+  });
+
+  // Give the server some time to start
+  tokio::time::sleep(tokio::time::Duration::from_secs(5)).await;
```

Comment:
If we don't spawn a new thread to the server we can await for tokio start instead of sleeping (which slow down automated tests)

Reply:
Does not work it gets stuck infinitely when awaiting directly

Reply:
We don't have a way to await the spawn then? What bothers me is having to sleep for 5 secs, this does not match the issue requirements of running under 100ms

Reply:
maybe this works?

```suggestion
  tokio::spawn(async {
    let file_paths = vec!["tests/server/config/server-start.graphql".to_string()];
    let config = Config::from_file_or_url(file_paths.iter()).await.unwrap();
    start_server(config).await.unwrap();
  })
  .await;
```

Reply:
Fixed it, needed to remove await from server.
Averaging at around 40 ms now
Thanks

Reply:
Nevermind that was flaky, I used a different approach